src/server/scripts/Northrend/Naxxramas/instance_naxxramas.cpp

```diff
@@ -431,6 +453,18 @@ class instance_naxxramas : public InstanceMapScript
         
         bool SetBossState(uint32 bossId, EncounterState state) override
         {
+            // pull all the trash if not killed
+            if (bossId == BOSS_PATCHWERK)
+                if (state == IN_PROGRESS)
+                    if (Creature* patch = instance->GetCreature(_patchwerkGUID))
+                        if (!PatchwerkRoomTrash.empty())
+                            for (std::list<uint64>::iterator itr = PatchwerkRoomTrash.begin(); itr != PatchwerkRoomTrash.end(); ++itr)
```

Comment:
Use auto here, its pretty obvious what this.

Reply:
wasn't sure ac supported c++11 code